vector/src/main/java/im/vector/Matrix.java

```diff
@@ -719,13 +740,52 @@ public void onRoomKeyRequestCancellation(IncomingRoomKeyRequestCancellation requ
                             KeyRequestHandler.getSharedInstance().handleKeyRequestCancellation(request);
                         }
                     });
+
+                    registerKeyBackupStateListener(session);
                 }
             }
         });
 
+
         return session;
     }
 
+    private void registerKeyBackupStateListener(MXSession session) {
+        if (session.getCrypto() != null) {
+            KeysBackup kb = session.getCrypto().getKeysBackup();
+            final String matrixID = session.getMyUserId();
+            if (keyBackupStateListeners.get(matrixID) == null) {
+                KeysBackupStateManager.KeysBackupStateListener keyBackupStateListener = new KeysBackupStateManager.KeysBackupStateListener() {
+                    @Override
+                    public void onStateChange(@NotNull KeysBackupStateManager.KeysBackupState newState) {
+                        if (KeysBackupStateManager.KeysBackupState.WrongBackUpVersion == newState) {
+                            //We should show the popup
+                            Activity activity = VectorApp.getCurrentActivity();
+                            //This is fake multi session :/ i should be able to have current session...
+                            if (activity != null) {
+                                new AlertDialog.Builder(activity)
+                                        .setTitle(R.string.new_recovery_method_popup_title)
+                                        .setMessage(R.string.new_recovery_method_popup_description)
+                                        .setPositiveButton(R.string.settings, (dialog, which) -> {
```

Comment:
Use `settings_troubleshoot_test_system_settings_quickfix` instead of creating a new key. Also I prefer to see verb in dialog action ("Open settings"), so it's better